arch/x86/kvm/pkvm/pgtable.c

```diff
@@ -269,10 +277,34 @@ static void pgtable_free_child(struct pkvm_pgtable_visit_ctx *ctx)
 	child = __pkvm_va(pgt_ops->pte_to_phys(ptep));
 	if (mm_ops->page_count(child) == 1) {
 		pgt_ops->pte_set(ptep, 0);
-		/* Flush TLB before release the child table page */
-		pgt_ops->flush_tlb(ctx->pgt, ctx->addr,
-				   pgt_ops->level_to_size(ctx->level));
-		mm_ops->put_page(child);
+		if (ctx->pgt->cap.flush_tlb_lazy) {
+			/*
+			 * As the flushing TLB is expensive, TLB will be flushed
+			 * when the walking is completed. In this case, the page
+			 * table page cannot be freed immediately at here as it
+			 * can be allocated as a page table page again before
+			 * flushing TLB. To prevent this, push the going-to-be
+			 * freed page table page to a teardown list and free
+			 * them after the TLB is flushed.
+			 */
+			list_add_tail((struct list_head *)child, &ctx->teardown_pages);
+			/*
+			 * The parent pte (pointed by the ctx->ptep) may still
```

Comment:
Maybe to make it clearer: "The old value of the parent pte"?

Reply:
yeah. It is better.